treeherder/etl/resultset_loader.py

```diff
@@ -31,8 +31,11 @@ def process(self, message_body, exchange):
             store_result_set_data(repo, [transformed_data])
 
         except ObjectDoesNotExist:
+            repo_info = (message_body["details"]
+                         if "details" in message_body
+                         else message_body["payload"])
```

Comment:
Good call.  I like your way better.  I've been doing JavaScript too much recently and forgot this obvious technique.  I feel like fixing the ``try..catch`` for ``ObjectDoesNotExist`` would be appropriate in this PR, since I'm hitting it.

I created https://bugzilla.mozilla.org/show_bug.cgi?id=1361570 for the index issue.